starter-code/app.js

```diff
@@ -1,86 +1,91 @@
-var express = require('express');
-var path = require('path');
-var favicon = require('serve-favicon');
-var logger = require('morgan');
-var cookieParser = require('cookie-parser');
-var bodyParser = require('body-parser');
-var app = express();
-
-var index = require('./routes/index');
-var users = require('./routes/users');
-const passportRouter = require("./routes/passportRouter");
-//mongoose configuration
-const mongoose = require("mongoose");
-mongoose.connect("mongodb://localhost/passport-local");
-//require the user model
-const User = require("./models/user");
-const session       = require("express-session");
-const bcrypt        = require("bcrypt");
-const passport      = require("passport");
-const LocalStrategy = require("passport-local").Strategy;
-const flash = require("connect-flash");
-
-
-
-
-
-//enable sessions here
-
-
-
-
-//initialize passport and session here
-
-
-
+// require npm packages
+const express = require('express');
+const path = require('path');
+const favicon = require('serve-favicon');
+const logger = require('morgan');
+const cookieParser = require('cookie-parser');
+const bodyParser = require('body-parser');
+const mongoose = require('mongoose');
+const session = require('express-session');
+const MongoStore = require('connect-mongo')(session);
+const flash = require('connect-flash');
+const passport = require('passport');
+const configurePassport = require('./helpers/passport');
+
+// run express
+const app = express();
+
+// require routes
+const index = require('./routes/index');
+const users = require('./routes/users');
+const passportRouter = require('./routes/passportRouter');
+
+// mongoose configuration
+mongoose.Promise = Promise;
+mongoose.connect('mongodb://localhost/ibi-ironhack', {
+  keepAlive: true,
+  reconnectTries: Number.MAX_VALUE
+});
 
+// enable sessions here
+app.use(session({
+  store: new MongoStore({
+    mongooseConnection: mongoose.connection,
+    ttl: 24 * 60 * 60 // 1 day
+  }),
+  secret: 'our-passport-local-strategy-app',
+  resave: true,
+  saveUninitialized: true,
+  cookie: {
+    maxAge: 24 * 60 * 60 * 1000
+  }
+}));
+
+// initialize passport and session
+configurePassport();// run before passport.initialize
+app.use(flash()); //
+app.use(passport.initialize());
+app.use(passport.session());
 
 // view engine setup
 app.set('views', path.join(__dirname, 'views'));
 app.set('view engine', 'ejs');
 
-
 app.use(logger('dev'));
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: false }));
 app.use(cookieParser());
 app.use(express.static(path.join(__dirname, 'public')));
-// require in the routers
+
+// -- middleware after passport init, before routes
+app.use((req, res, next) => {
+  res.locals.user = req.user; // -> what is this doing?
```

Comment:
When passport make a login, it creates a `req.user` with the user's id. So here were are saving this property in the locals for having as a "global variable" for using in the templates .